### PR TITLE
Update rapidjson to latest master

### PR DESCRIPTION
**Pull request by bkotzz**
This just merges in latest master from the tencent repo to the hunter-1.1.0 branch. Upstream hasn’t had a release in a while but there’s been a fair amount of commits. @ruslo what’s your thoughts on this? I can dig deeper to make sure all the tests still pass, but are we okay with updating the fork to latest master? And if so, should we make a different branch or use this one?

Update: Rebases `hunter-1.1.0` over `hunter-66eb606`, then fixes a couple merge conflicts.

### CODE REVIEW

**Comment by ruslo** [comment]
Branch `hunter-66eb606` created. @bkotzz please rebase it in such way that we don't have merge commits.

**Comment by bkotsopoulossc** [comment]
Will do, thanks

File: CMakeLists.txt [hunk 1]
```diff
@@ -59,14 +66,16 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     if(RAPIDJSON_ENABLE_INSTRUMENTATION_OPT AND NOT CMAKE_CROSSCOMPILING)
         if(CMAKE_SYSTEM_PROCESSOR STREQUAL "powerpc" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "ppc64" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "ppc64le")
           set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mcpu=native")
-        else()
+        elseif(NOT ANDROID)
```
Comment (bkotzz) [inline CMakeLists.txt:69]: I added this to handle https://github.com/Tencent/rapidjson/commit/ffe3a3daa80d49ada0a786225d1935048e4e5ada. Note that the `-W*` flags are already set below for all.

File: CMakeLists.txt [hunk 2]
```diff
@@ -59,14 +66,16 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
...
           #FIXME: x86 is -march=native, but doesn't mean every arch is this option. To keep original project's compatibility, I leave this except POWER.
           set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native")
         endif()
     endif()
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror")
     set(EXTRA_CXX_FLAGS -Weffc++ -Wswitch-default -Wfloat-equal -Wconversion -Wsign-conversion)
     if (RAPIDJSON_BUILD_CXX11 AND CMAKE_VERSION VERSION_LESS 3.1)
+        # If cmake version < 3.1, no support for CMAKE_CXX_STANDARD (set above),
```
Comment (bkotzz) [inline CMakeLists.txt:77]: Adding this comment so it is clear why this is done, as there is already a line `set(CMAKE_CXX_STANDARD 11)` above

**Review by ruslo — APPROVED** [review]

**Comment by ruslo** [comment]
Released: https://github.com/hunter-packages/rapidjson/releases/tag/v1.1.0-66eb606-p0

**Pull request merged**